### ISSUE TITLE
Shortcuts not unregisted on minify event

### ISSUE THREAD

**Issue by jamesrobb** [parro-it/electron-localshortcut#16]
Hello,

I was having an issue with the electron application Caprine posted here:
https://github.com/sindresorhus/caprine/issues/202

The issue was that f5  and ctrl+shift+c were not being unregistered when I minimized Caprine to the tray, so I could no longer use f5 to refresh pages in my browser; the key was still bound to Caprine.

If I brought Caprine out of the tray or killed the application, I regained control of the f5 key.

I have been told that the issue is likely with this package not unregistering the shortcuts on a minify event.

Is there something I can do to help confirm/fix this?

**Comment by parro-it** [comment]
Thank you for the report @jamesrobb, I will check it soon

**Comment by parro-it** [comment]
I implemented the logic for the `minimize`/`restore` and `show`/`hide` events in https://github.com/parro-it/electron-localshortcut/commit/fdd0fcde5655168444f1d3c86824c52f0589d537.

I have some doubt on what version should I publish with on npm. This could be considered a breaking change, since some app could be relying on a shortcut being working even with a minimized window.

What do you think?

/cc @sindresorhus

**Comment by sindresorhus** [comment]
I doubt anyone depends on that, but just do a major release to be safe.

**Comment by zeke** [comment]
@marshallofsound does this potentially address the concerns you had with this module?

**Comment by parro-it** [comment]
Just published v2.0.0 that implements the logic for the minimize/restore and show/hide events.
@marshallofsound, @zeke, what are these concerns you are talking about? Could I be of some help?

**Comment by zeke** [comment]
> what are these concerns you are talking about?

See https://github.com/electron/electron/pull/8775/files/ba4a2d7c307cc5fe9e57aab898f6b619babc1f81#r103096101

**Comment by parro-it** [comment]
Thank you @zeke, I think these changes could at least alleviate some of the issue he was talking about. But I'm not sure it can solve them completely. I have to test what happens when you register a shortcut already registered as global.

**Comment by parro-it** [comment]
@zeke I tested the behavior of version 2.0.0 in regards of conflicting shortcuts.

Actually, if you register with electron-localshortcut a shortcut that is already registered by the OS or by another app, your shortcut take precedence on the other one. 

But only when your window is not minified and has focus. When you minimize your window, the other shortcut is correctly restored.

Instead, if you listen to key event in the DOM, the shortcut of the other app take always precedence, even when your window is focused.

I think sometime this behavior could be desirable and sometime not. Anyway, I saw @MarshallOfSound was suggesting an alternative DOM implementation to solve the problem.

I'm open to implement it if there is interest about...
